packages/openapi-ts/src/utils/write/transformers.ts

```diff
@@ -0,0 +1,328 @@
+    // the transformer was never generated
+    unsetUniqueTypeName({
+      client,
+      name,
+    });
+    onRemoveNode?.();
+    return result;
+  }
+
+  const expression = compiler.types.function({
+    multiLine: true,
+    parameters: [
+      {
+        name: dataVariableName,
+      },
+    ],
+    statements: [
+      ...statements,
+      compiler.return.statement({
+        expression: ts.factory.createIdentifier(dataVariableName),
+      }),
+    ],
+  });
+  const statement = compiler.export.const({
+    expression,
+    name,
+    typeName: name,
+  });
+  onNode(statement);
+
+  return {
+    created: true,
+    name,
+  };
+};
+
+export const processResponseTransformers = async ({
+  client,
+  onNode,
+  onRemoveNode,
+}: Pick<TypesProps, 'client' | 'onNode' | 'onRemoveNode'>) => {
+  const config = getConfig();
+
+  for (const service of client.services) {
+    for (const operation of service.operations) {
+      const hasRes = operation.results.length;
+
+      if (!hasRes) {
+        continue;
+      }
+
+      const responses = getSuccessResponses(operation.results);
+
+      const nonVoidResponses = responses.filter(
+        (response) => !isVoidResponse(response),
+      );
+
+      if (!nonVoidResponses.length) {
+        continue;
+      }
+
+      if (nonVoidResponses.length > 1) {
+        if (config.debug) {
+          console.warn(
+            `⚠️ Transformers warning: route ${operation.method} ${operation.path} has ${responses.length} success responses. This is currently not handled and we will not generate a response transformer. Please open an issue if you'd like this feature https://github.com/hey-api/openapi-ts/issues`,
+          );
+        }
+        continue;
+      }
+
+      const name = operationResponseTypeName(operation.name);
+      generateType({
+        client,
+        meta: {
+          $ref: `transformers/${name}`,
+          name,
+        },
+        nameTransformer: operationResponseTransformerTypeName,
+        onCreated: (nameCreated) => {
+          const statements =
+            responses.length > 1
+              ? responses.flatMap((response) => {
+                  const statements = processModel({
+                    client,
+                    meta: {
+                      $ref: `transformers/${name}`,
+                      name,
+                    },
+                    model: response,
+                    onNode,
+                    onRemoveNode,
+                    path: [dataVariableName],
+                  });
+
+                  // assume unprocessed responses are void
+                  if (!statements.length) {
+                    return [];
+                  }
+
+                  return [
+                    compiler.logic.if({
+                      expression: compiler.logic.safeAccess(['data']),
+                      thenStatement: ts.factory.createBlock(statements),
+                    }),
+                  ];
+                })
```

Comment:
What happens if there's an endpoint with two different models on responses? Looks like this attempts to handle it but there's no test?

Reply:
Do you have an example? It's meant to handle only `FooResponse | void` and there's a snapshot for that

Reply:
Ah you're right, i missed that you're still bailing out for 2+ non-void models.

Should be easy to adapt to support multiple responses when one of us has time for